etest/profile.py

```diff
@@ -0,0 +1,182 @@
+    armv6 = "armv6j_hardfp"
+    armv7 = "armv7a_hardfp"
+    arm64 = "arm64"
+
+
+class InvalidProfileError(ValueError):
+    """The profile is invalid."""
+
+    pass
+
+
+class Profile:
+    """A class to generate profile options."""
+
+    def __init__(
+        self,
+        strict: bool,
+        arch: str,
+        libc: str,
+        hardened: bool,
+        multilib: bool,
+        systemd: bool,
+    ) -> None:
+        """Initialize parameters."""
+        self._strict = strict
+
+        self._arch = arch
+        self._docker_arch = arch
+        self._pkg_arch = arch
+
+        self._libc = libc
+        self._hardened = hardened
+
+        self._multilib = multilib
+
+        self._systemd = systemd
+
+        _LOGGER.info("Standarizing profile data.")
+        self._standarize()
+
+    @property
+    def strict(self) -> bool:
+        """Error on warnings."""
+        return self._strict
+
+    @property
+    def arch(self) -> str:
+        """Architecture to use."""
+        return self._arch
+
+    @property
+    def docker_arch(self) -> str:
+        """Architecture for the docker base image."""
+        return self._docker_arch
+
+    @docker_arch.setter
+    def docker_arch(self, value: str) -> None:
+        self._docker_arch = value
+
+    @property
+    def pkg_arch(self) -> str:
+        """Gentoo package architecture."""
+        return self._pkg_arch
+
+    @pkg_arch.setter
+    def pkg_arch(self, value: str) -> None:
+        self._pkg_arch = value
+
+    @property
+    def libc(self) -> str:
+        """Libc to be used."""
+        return self._libc
+
+    @property
+    def hardened(self) -> bool:
+        """Use hardening."""
+        return self._hardened
+
+    @property
+    def multilib(self) -> bool:
+        """Use multilib."""
+        return self._multilib
+
+    @multilib.setter
+    def multilib(self, value: bool) -> None:
+        self._multilib = value
+
+    @property
+    def systemd(self) -> bool:
+        """Use systemd."""
+        return self._systemd
+
+    @functools.cached_property
+    def base_profile(self) -> str:
+        """Build profile information."""
+        base_profile = ""
+
+        if self.libc != "glibc":
+            base_profile += f"-{self.libc}"
+
+        if self.hardened:
+            base_profile += "-hardened"
+        elif self.libc != "glibc":
+            base_profile += "-vanilla"
+
+        if not self.multilib:
+            base_profile += "-nomultilib"
+
+        if self.systemd:
+            base_profile += "-systemd"
+
+        return base_profile
+
+    @property
+    def profile(self) -> str:
+        """Profile to use."""
+        return self.arch + self.base_profile
+
+    @property
+    def docker_profile(self) -> str:
+        """Profile to use for the base image."""
+        return self.docker_arch + self.base_profile
+
+    def _warn(self, message: str) -> None:
+        """Give a warning to the user."""
+        if not self.strict:
```

Comment:
When not using strict, it will emit a warning when using `--no-multilib` on an architecture other than amd64. This warning just lets the user know it will not do anything. It doesn't affect the safety of the program.

Reply:
I'd say we always want that warning.  There is never a reason to skip warning about something that doesn't take action.  In fact, I'd argue that case should throw an error instead of just warn.